dockerfiles/git.Dockerfile

```diff
@@ -0,0 +1,24 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.18
+ARG ALPINE_VERSION=3.16
+ARG GIT_VALIDATION_VERSION=v1.1.0
+
+FROM alpine:${ALPINE_VERSION} AS base
+RUN apk add --no-cache git gpg
```

Comment:
I realise plain alpine is smaller than `golang:alpine`, but wondering if this actually helps;

- we already need  `golang:alpine` (see below)
- pulling `alpine:<some version>` may mean that the version of alpine is different than the `golang:alpine` version, so we wouldn't be sharing layers (and pulling two distinct images).

I'm included to say; just use `golang:<version>-alpine` for both, as they'll always (or _most of the time_) be run in tandem, so would save us pulling two images?

Reply:
Oh! Actually, so we also use `alpine` for the main dockerfile; https://github.com/distribution/distribution/blob/87941225ed8914a606d648783d0f475924157997/Dockerfile#L52

So alternatively;

- change the main Dockerfile to also have a `ARG ALPINE_VERSION=3.16`
- change all `golang:${GO_VERSION}-alpine` images to use the `ALPINE_VERSION` as suffix (`golang:${GO_VERSION}-alpine${ALPINE_VERSION}`) that way we know the Golang image is using the same Alpine version (so layers _should_ be shared); at least it prevents any potential issues where we build on a different version of alpine as we ship on (final stage of the main Dockerfile) 
- having `ALPINE_VERSION` in both Dockerfiles helps with grep'ing if someone wants to update (and also allow overriding the version through build-args)